core/trino-main/src/main/java/io/trino/server/protocol/spooling/SpoolingManagerBridge.java

```diff
@@ -87,54 +88,68 @@ public boolean useInlineSegments()
     @Override
     public SpooledSegmentHandle create(SpoolingContext context)
     {
-        return delegate().create(context);
+        try (ThreadContextClassLoader _ = new ThreadContextClassLoader(delegate().getClass().getClassLoader())) {
```

Comment:
This should be up to the implementation. We don't do this in the engine for other plugins. Instead, there are wrappers such as `ClassLoaderSafeEventListener`.

Also, why is this needed? We shouldn't have class loader dependent code anymore.